File: AtCoder/abc/166d.py
X = int(input())

for b in range(-10**6, 10**6):
    a5 = abs(b**5 + X)
    a = int(pow(a5, 1 / 5))

    if a5 < 0:
        a = -a

    if a**5 - b**5 == X:
        print(a, b)
        exit()

File: AtCoder/abc/166d.rs
#![allow(unused_imports)]
#![allow(unused_macros)]
#![allow(non_snake_case)]
#![allow(dead_code)]
#![allow(unused_variables)]
#![allow(unused_mut)]
#![allow(unused_assignments)]
use proconio::input;
use proconio::marker::Usize1;
use std::collections::*;
use std::cmp::*;
use std::f64::consts::*;

const MOD: u64 = 1000000007;
const INF: usize = std::usize::MAX / 4;

fn main() {
    input! {
        X: i64,
    }

    for a in -500_i64..501_i64 {
        for b in -500_i64..501_i64 {
            if a.pow(5) - b.pow(5) == X {
                println!("{} {}", a, b);
                return;
            }
        }
    }

}
